app/javascript/components/recipes/edit/recipe-steps.vue

```diff
@@ -66,8 +100,23 @@ export default {
     return {
       newStepDescription: null,
       error: '',
+      modifiedRecipeSteps: [],
+      recipeStepsSet: new Set(),
     };
   },
+  watch: {
+    recipeSteps() {
+      for (let idx = 0; idx < this.recipeSteps.length; idx++) {
+        if (!this.recipeStepsSet.has(this.recipeSteps[idx])) {
+          this.modifiedRecipeSteps.push({
+            step: this.recipeSteps[idx],
+            isEditing: false,
+          });
+          this.recipeStepsSet.add(this.recipeSteps[idx]);
+        }
+      }
+    },
```

Comment:
- Ojo que los watch no se llaman al montar el componente, solo frente a cambios de la prop (aunque hay una opción para que se ejecuten al inicializarse - [acá](https://forum.vuejs.org/t/watchers-not-triggered-on-initialization/12475/10) y [acá](https://vuejs.org/v2/api/#watch) más info). Me imagino que si ahora les está funcionando, el componente debe estar recibiendo primero unas prop vacías y luego cambia al arreglo de steps, algo así o no? Cuidado igual ahí, consideraría un valor inicial, o usar `immediate` 
- El set me suena que no es muy necesario. Podrías buscar directamente en el `modifiedRecipeSteps` si existe ya el step

Reply:
- Exacto, la componente se inicializa vacía, pero se hace un GET de la receta y ahí se actualiza. Igualmente, le voy a dar un valor inicial de "No encontramos pasos en tu receta".
- Se podría decir que el set está demás, lo agregué sólo para ahorrarme unas líneas extras, además que creo que es más intuitivo.